### PR TITLE
DR-11 Add specification README.

### PR DESCRIPTION
**Pull request by filipblnt**
Added description of product specification. Reorganised repo structure wrt specifications.

Ref: https://github.com/blindnet-io/devrel-management/issues/11

### CODE REVIEW

File: specifications/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Specification
+
+The first step of building products at blindnet always includes specification writing. It helps not only us to faster build better products, but we hope it also helps you to better and faster understand how our products work.
```
Comment (m4rk055) [inline specifications/README.md:3]:
```suggestion
The first step of building products at blindnet always includes specification writing. It helps not only us to rapidly build better products, but we hope it also helps others better understand how our products work.
```

File: specifications/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+Functional requirements define functional aspects of the product and are captured in the functional requirements document (FRD). Each functional requirement defines one particular function a product needs to deliver, and is assigned a unique code identifier (e.g., FRD-BE01, FR-SDK14). 
+
+### Architecture
+
+Product architecture defines different components of the product structure and how they interact together in order to deliver specified functions. For example, the architecture document defines different SDKs, APIs and databases that, when put together, deliver promised functional aspects. 
+
+### Design
+
+Product design defines in detail the technical aspects of each architectural component of the product. It includes, for example, database models, or expected inputs/outputs of SDK functions and REST API routes. In some cases, like for example for REST API routes, the design is described together with the corresponding functional requirement.
```
Comment (m4rk055) [inline specifications/README.md:40]:
```suggestion
Product design defines in detail the technical aspects of each architectural component of the product. It includes, for example, database models, or expected inputs/outputs of SDK functions and REST API endpoints. In some cases, like for example for REST API endpoints, the design is described together with the corresponding functional requirement. 
```

**Comment by filipblnt** [comment]
All comments are now addressed.

**Comment by noelmace** [comment]
LGTM. As you assigned this PR to me, I suppose you're expecting me to merge it.

**Pull request merged**